public/app/plugins/datasource/prometheus/legend.ts

```diff
@@ -2,6 +2,39 @@ import { Labels } from '@grafana/data';
 
 /** replace labels in a string.  Used for loki+prometheus legend formats */
 export function renderLegendFormat(aliasPattern: string, aliasData: Labels): string {
-  const aliasRegex = /\{\{\s*(.+?)\s*\}\}/g;
+  // \{\{ Escaped "{{" characters (char code 123).
+  //
+  // \s*  Matches any whitespace character (spaces, tabs, line breaks) 0 or more times.
+  //
+  // (?!  Negative lookahead. Specifies a group that can not match after the main
+  //      expression (if it matches, the result is discarded).
+  //
+  //        \s Whitespace. Matches any whitespace character (spaces, tabs, line breaks).
+  //
+  // )
+  //
+  // (    Capturing group #1. Groups multiple tokens together and creates a capture group for extracting a
+  //      substring or using a backreference.
+  //
+  //      [^ Match a single character not present in the list below
+  //           \{ matches the character { (char code 123)
+  //           \} matches the character } (char code 123)
+  //      ]
+  //        + Quantifier. Match 1 or more of the preceding token.
+  //        ? Lazy. Makes the preceding quantifier lazy, causing it to match as few characters as possible.
+  //
+  // )
+  //
+  // (?!  Negative lookbehind. Specifies a group that can not match before the main
```

Comment:
One more thing - negative lookbehind is not supported in safari :( https://caniuse.com/js-regexp-lookbehind

Reply:
good catch! i'll close this one since @kaydelaney's PR seems like the better approach to handle the `\s` given that info